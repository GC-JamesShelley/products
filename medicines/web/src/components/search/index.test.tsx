import { shallow } from 'enzyme';
import React from 'react';
import Search from './index';

const onSearchBlurMock = jest.fn();
const onSearchChangeMock = jest.fn();
const onSearchSubmitMock = jest.fn();
const searchMock = 'drug';

describe(Search, () => {
  it('should render', () => {
    const component = shallow(
      <Search
        onSearchBlur={onSearchBlurMock}
        onSearchChange={onSearchChangeMock}
        onSearchSubmit={onSearchSubmitMock}
        search={searchMock}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
