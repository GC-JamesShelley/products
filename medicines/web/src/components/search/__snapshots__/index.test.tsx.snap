// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render 1`] = `
<styled.section>
  <styled.h3>
    Search by text
  </styled.h3>
  <form
    aria-label="Search for medicines"
    onSubmit={[MockFunction]}
    role="search"
  >
    <label
      htmlFor="search"
    >
      Enter a product, active substance, or PL number:
    </label>
    <div
      className="searchbar"
    >
      <input
        id="search"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        type="search"
        value="drug"
      />
      <styled.input
        type="submit"
        value="Search"
      />
    </div>
  </form>
</styled.section>
`;
